renku/api/datasets.py

```diff
@@ -150,32 +151,36 @@ def add_data_to_dataset(
                     dataset, dataset_path, url, target, **kwargs
                 )
             else:
-                files = {}
+                files = []
                 for t in target:
-                    files.update(
+                    files.extend(
```

Comment:
Why not append? All return lists are of length 1. Maybe instead of allocating additional lists and then merging them, we could just append the object?

Reply:
if there are subdirectories it will not be length 1